editor/sidebar/post-taxonomies/hierarchical-term-selector.js

```diff
@@ -221,6 +224,7 @@ class HierarchicalTermSelector extends Component {
 							placeholder={ newTermLabel }
 							value={ formName }
 							onChange={ this.onChangeFormName }
+							required
```

Comment:
On principle I'd be fine with the browser default behavior, but if we do decide to trim the input, we might want to consider how to convey this to the user, since the `required` browser default won't appear. Alternatively, we could prevent whitespace from being added to the `value` if no other non-whitespace characters in value.